### DISCUSSION TITLE
Setting collision shape on Initialize() func does not work as expected

### DISCUSSION THREAD

**Discussion by ullanar** [quaadgras/graphics.gd#50] [Is this a bug?]
Hello! trying to initialize class with collision shape and using `Initialize` func create shape for it

If I understand correctly that code should create class with filled collision shape

```go
type InteraciveArea struct {
	classdb.Extension[InteraciveArea, Area2D.Instance]
	CollisionShape CollisionShape2D.Instance
}

func (i *InteraciveArea) Initialize() {
	shape := CircleShape2D.New()
	i.CollisionShape.SetShape(shape.AsShape2D())
}
```

But it does not seems that case 
![image](https://github.com/user-attachments/assets/99825db8-5571-4646-a931-49297589da8e)
![image](https://github.com/user-attachments/assets/5e53839f-a743-4506-a547-cc229c23b7f1)

Did I miss smth or this is a bug?

**UPD**: It seems like `Initialize()` method even was not called. Still think what I miss something, can you suggest a solution?

**Comment by Splizard** [comment]
You'll want to be using a `Ready` function (equivalent to GDScript `_ready`). Initialize can only be implemented by MainLoop classes.
